packages/core/package.json

```diff
@@ -20,6 +20,10 @@
     "./lib/message/topic_only_message": {
       "types": "./dist/lib/message/topic_only_message.d.ts",
       "import": "./dist/lib/message/topic_only_message.js"
+    },
+    "./lib/base_protocol": {
```

Comment:
We are getting more of these `exports`, maybe we can split some of the into separate packages?

Reply:
Hum, the exports are about exposing some implementation specific code that we would not want a library consumer to use (hence don't want to export from index.js).

Not sure it warrants a separate package. What is your decision making process in terms of extracting to another package?

Reply:
Well, exposing it means it will potentially be used by consumers (and in some examples we show usage of it)

I think we should extract things to a package if they correlate with a specific topic. Here I see we are exposing it because we want to use it in another package, that's essentially `protected` property, I would move such a thing to a package and make it private - not sure how it will work out in the end

Reply:
We can't really make it private tho. can we?

Reply:
I think we can make it private to avoid publishing and keep stuff in it. I can check how it would look like tomorrow or on Friday.

Reply:
yes but this base protocol package would still be a dependency of waku core. When a user runs `npm i @waku/core` it will not find base protocol package in npm because private and hence install will fail, right?

Reply:
I am not able to definitely answer it so that's why I need to check it more, but I think that it is possible to bypass it and bundle such packages within.